src/basic/pidref.c

```diff
@@ -260,6 +260,13 @@ int pidref_verify(PidRef *pidref) {
         return 1; /* We have a pidfd and it still points to the PID we have, hence all is *really* OK → return 1 */
 }
 
+
+int pidref_is_alive(PidRef *pidref) {
```

Comment:
Hmm, I find this difficult to implement. `pidref_is_alive()` requires prototypes of `pidfd_is_alive()` and `pid_is_alive()`. However, if I include `process-util.h` in `pidref.h` I am getting include loop. If I define these prototypes manually then I am getting a ton of warnings because we compile with `-Wredundant-decls`.

Reply:
don't sweat it I'd say ;-)

Reply:
(but I'd put some more work on you, sorry: if you add pidref_is_alive() please move relevant candidates over to using it. Specifically in src/core/service.c there are two invocations if pid_is_alive() you can trivially switch over to pidref_is_alive(). once you did that, please remove the line that says `- pid_is_alive() → pidref_is_alive()` in TODO, since you moved one more thing over, successfully).

Reply:
Will do.